### PR TITLE
Present an alert for setting a new password

### PR DESCRIPTION
**Pull request by crazytonyli**
> [!Note]
> This PR is built on top of https://github.com/wordpress-mobile/WordPress-iOS/pull/23777.

Changing from pushing a separate page to presenting an alert.

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift [hunk 1]
```diff
@@ -62,15 +69,34 @@ struct UserDetailsView: View {
...
+                    Task {
+                        try await self.actionDispatcher.setNewPassword(id: user.id, newPassword: newPassword)
+                    }
+                }
+                .disabled(newPassword.isEmpty || newPassword != newPasswordConfirmation)
+                Button(role: .cancel) {
+                    presentPasswordAlert = false
+                } label: {
+                    // TODO: Replace with `SharedStrings.Button.cancel`
+                    Text(NSLocalizedString("shared.button.cancel", value: "Cancel", comment: "A shared button title used in different contexts"))
```
Comment (kean) [inline Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift:88]: would've probably been faster to do it that to add a TODO :laugh

File: Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift [hunk 2]
```diff
@@ -120,23 +146,35 @@ struct UserDetailsView: View {
...
             comment: "The 'Delete User' button on the user profile – matches what's in /wp-admin/profile.php"
         )
-    }
-}
 
-private struct UserDetailLabeledContentStyle: LabeledContentStyle {
-    func makeBody(configuration: LabeledContentStyleConfiguration) -> some View {
-        LabeledContent(configuration)
+        static let newPasswordAlertMessage = NSLocalizedString(
+            "userdetail.newPasswordAlertMessage",
```
Comment (kean) [inline Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift]: should be `userDetails.*`

**Review by kean — APPROVED** [review]
Approve with two minor comments.

File: Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift (hunk 1 above)
Comment (crazytonyli) [reply Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift:88]: Need to move all the "Users" code to the app target first 🥲

Comment (kean) [reply Modules/Sources/WordPressUI/Views/Users/Views/UserDetailsView.swift:88]: Ah, right. Yeah, I'd suggest moving it until there is an idea of how to organize it better.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**